### PR TITLE
Chore: Code cleanup

### LINKED ISSUE #2224: Improve code quality from the Sonar Cloud stats
**Describe the bug**
There are currently multiple code smells, bugs, and security hotspots. Some of these can be solved to improve the code quality

**Screenshots**
image

### PR DESCRIPTION
**Pull request by ElinorW**
## Overview

Reduce code smells and bugs from sonar cloud analysis
Fixes #2224 

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

List of code smells addressed in this PR

- [x] Remove TODO comment
- [x] "Switch" statements should have at least 3 cases
- [x] Switch case should end with an unconditional break, continue, return or throw statement
- [x] Cases in switch statements have repeated code blocks 
- [x] Nested ternary operations
- [x] Loop counters being modified in the loop body
- [x] Use of deprecated methods such as substr, addListener, removeListener
- [x] React components must start with an uppercase.
- [x] Redundant assignments
- [x] A shorthand property defined after a longhand property will completely override the value defined in the longhand property making the longhand one useless.
- [x] Unused imports
- [x] Using `?` and `undefined` together as they both communicate the same thing

## Testing Instructions

* How to test this PR
* Prefer bulleted description
* Start after checking out this branch
* Include any setup required, such as bundling scripts, restarting services, etc.
* Include test case, and expected output

### CODE REVIEW

**Comment by thewahome** [comment]
Not really important, but as you go along, which code smells have been closed? It would be a good learning experience for us to see what the problem was and how we've fixed it

**Comment by ElinorW** [comment]
> Not really important, but as you go along, which code smells have been closed? It would be a good learning experience for us to see what the problem was and how you've fixed it. Maybe indexing them in the Notes section of the overview

Ah yes, this is great feedback.
I shall reference them in the notes section as requested

File: src/app/views/App.tsx [hunk 1]
```diff
@@ -86,7 +86,7 @@ class App extends Component<IAppProps, IAppState> {
 
   public componentDidMount = async () => {
     this.displayToggleButton(this.mediaQueryList);
-    this.mediaQueryList.addListener(this.displayToggleButton);
+    this.mediaQueryList.addEventListener('click',this.displayToggleButton);
```
Comment (thewahome) [inline src/app/views/App.tsx:89]: Using devtools I can set browser to different phones. 
What I have noticed is that the app doesn't know that it's in mobile view so it does not respond appropriately.

I think the culprit could be setting the listener to listen to a 'click' event. I am not sure which one it should be.

**Review by thewahome — CHANGES_REQUESTED** [review]

File: src/app/views/App.tsx (hunk 1 above)
Comment (ElinorW) [reply src/app/views/App.tsx:89]: Ah, this is a great observation. 
Okay, let me look this up... 
There's probably a more appropriate listener

**Review by thewahome — APPROVED** [review]

**Review by Onokaev — APPROVED** [review]

**Pull request merged**